pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs

```diff
@@ -506,14 +546,62 @@ private void UseSavedColumnsIfValid()
                 SetColumnPositions(detectedColumns);
             }
         }
+
         /// <summary>
         /// Returns the current column positions as a list of strings
         /// </summary>
-        private List<string> CurrentColumnPositions()
+        public List<string> CurrentColumnPositions()
         {
             return ComboBoxes.Select(combo => combo.Text).ToList();
         }
 
+        /// <summary>
+        /// Returns the current column positions as a list of strings in invariant language
+        /// </summary>
+        public List<string> CurrentColumnPositionsInvariant()
+        {
+            return ColumnNamesInvariant(CurrentColumnPositions());
+        }
+        public static List<string> ColumnNamesInvariant(IEnumerable<string> local)
+        {
+            return local.Select(ColumnNameInvariant).ToList();
+        }
+
+        #region testing support
+
+        /// <summary>
+        /// For testing purposes only - normally this is done one at a time in UI rather than programatically
+        /// </summary>
+        public void SetSelectedColumnTypes(params string[] headers)
+        {
+            var index = 0;
+            foreach (var header in headers)
+            {
+                if (!string.IsNullOrEmpty(header) && ComboBoxes.Count > index)
+                {
+                    ComboBoxes[index].SelectedIndex = ComboBoxes[1].FindStringExact(header);
+                }
+                index++;
+            }
+        }
+
+        public static string ColumnNameInvariant(string local)
+        {
+            var headerTypesInvariant = ImportTransitionListColumnSelectDlg.GetKnownHeaderTypesInvariant();
+            var headerTypes = ImportTransitionListColumnSelectDlg.GetKnownHeaderTypes();
+            var index = headerTypes.IndexOf(s => s.Item1.Equals(local));
+            if (index < 0)
+            {
+                index = headerTypesInvariant.IndexOf(s => s.Item1.Equals(local));
+            }
+            return headerTypesInvariant[Math.Max(0, index)].Item1;
+        }
+        
+        public string[] SupportedColumnTypes => Enumerable.Range(0, ComboBoxes[1].Items.Count - 1).Select(i => ComboBoxes[1].Items[i].ToString()).ToArray();
```

Comment:
Much tidier, thanks. I think it can be made even more compact:
SupportedColumnTypes => ComboBoxes[1].Items.Select(value=>value.ToString()).ToArray();